File: IdeaProjects/Java/Infant mathematics assisted learning system/Employee information management system/src/EmployeeManagementSystem.java

import java.io.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

// 员工管理系统类
public class EmployeeManagementSystem {
    private List<Employee> employees =new ArrayList<>(); // 员工列表
    Scanner scanner = new Scanner(System.in);

    // run方法，用于运行员工管理系统
    public void run() {
        // 循环处理用户的操作选择
        boolean isRunning = true;
        while (isRunning) {
            System.out.println("请选择操作：");
            System.out.println(" 1 ：录入职工信息");
            System.out.println(" 2 ：显示所有职工信息");
            System.out.println(" 3 ：查询职工信息");
            System.out.println(" 4 ：删除职工信息");
            System.out.println(" 5 ：修改职工信息");
            System.out.println(" 6 ：保存数据到文件");
            System.out.println(" 7 ：从文件打开数据");
            System.out.println(" 0 ：退出系统");

            System.out.println("请输入你的操作：");
            int choice = scanner.nextInt();
            scanner.nextLine(); // 清除输入缓冲区的换行符

            // 根据用户的选择，执行对应的操作
            switch (choice) {
                case 0 -> isRunning = false;
                case 1 -> addEmployee();
                case 2 -> displayAllEmployees();
                case 3 -> searchEmployee();
                case 4 -> deleteEmployee();
                case 5 -> updateEmployee();
                case 6 -> saveToFile();
                case 7 -> openFromFile();
                default -> System.out.println("无效的选择！");
            }
        }
    }

    // 员工管理系统构造函数
    public EmployeeManagementSystem() {

    }

    // 添加员工
    public void addEmployee() {
        System.out.println("开始录入职工信息");
        System.out.println("请输入职工号：");
        String id = scanner.nextLine();

        for (Employee employee : employees) {
            if (employee.getId().equals(id)) {
                System.out.println("职工号已存在！");
                return;
            }
        }

        System.out.println("请输入姓名：");
        String name = scanner.nextLine();
        System.out.println("请输入性别：");
        String gender = scanner.nextLine();
        System.out.println("请输入出生年月：");
        String birthDate = scanner.nextLine();
        System.out.println("请输入学历：");
        String education = scanner.nextLine();
        System.out.println("请输入职务：");
        String position = scanner.nextLine();
        System.out.println("请输入工资：");
        double salary = scanner.nextDouble();
        scanner.nextLine(); // 清除输入缓冲区的换行符
        System.out.println("请输入住址：");
        String address = scanner.nextLine();
        System.out.println("请输入电话：");
        String phoneNumber = scanner.nextLine();

        Employee employee = new Employee(id, name, gender, birthDate, education, position, salary, address, phoneNumber);
        employees.add(employee);
        System.out.println("职工信息录入成功！");
        System.out.println();
    }

    // 显示所有员工信息
    public void displayAllEmployees() {
        System.out.println("所有职工信息如下：");
        for (Employee employee : employees) {
            employee.displayInfo();
            System.out.println();
        }
        System.out.println();
    }

    // 搜索员工
    public void searchEmployee() {
        System.out.println("请输入要查询的职工号：");
        String id = scanner.nextLine();

        for (Employee employee : employees) {
            if (employee.getId().equals(id)) {
                employee.displayInfo();
                System.out.println();
                return;
            }
        }
        System.out.println("未找到该职工");
        System.out.println();
    }

    // 删除员工
    public void deleteEmployee () {
            System.out.println("请输入要删除的职工号：");
            String id = scanner.nextLine();

            for (Employee employee : employees) {
                if (employee.getId().equals(id)) {
                    employees.remove(employee);
                    System.out.println("职工删除成功");
                    System.out.println();
                    return;
                }
            }
            System.out.println("未找到该职工");
            System.out.println();
        }

    // 更新员工信息
    public void updateEmployee (){
            System.out.println("请输入要修改的职工号：");
            String id = scanner.nextLine();

            for (Employee employee : employees){
                if (employee.getId().equals(id)){
                    System.out.println("请选择要修改的信息：");
                    System.out.println("1.职工号");
                    System.out.println("2.姓名");
                    System.out.println("3.性别");
                    System.out.println("4.出生年月");
                    System.out.println("5.学历");
                    System.out.println("6.职务");
                    System.out.println("7.工资");
                    System.out.println("8，住址");
                    System.out.println("9.电话");

                    int choice= scanner.nextInt();
                    scanner.nextLine(); // 清除输入缓冲区的换行符

                    switch (choice){
                        case 1 ->{
                            System.out.println("请输入职工号：");
                            String newId = scanner.nextLine();
                            employee.setId(newId);
                            System.out.println("信息修改成功！");
                            System.out.println();
                            return;
                        }
                        case 2 ->{
                            System.out.println("请输入姓名：");
                            String newName = scanner.nextLine();
                            employee.setName(newName);
                            System.out.println("信息修改成功！");
                            System.out.println();
                            return;
                        }
                        case 3 ->{
                            System.out.println("请输入性别：");
                            String newGender = scanner.nextLine();
                            employee.setGender(newGender);
                            System.out.println("信息修改成功！");
                            System.out.println();
                            return;
                        }
                        case 4->{
                            System.out.println("请输入出生年月：");
                            String newBirthDate = scanner.nextLine();
                            employee.setBirthDate(newBirthDate);
                            System.out.println("信息修改成功！");
                            System.out.println();
                            return;
                        }
                        case 5->{
                            System.out.println("请输入学历：");
                            String newEducation = scanner.nextLine();
                            employee.setEducation(newEducation);
                            System.out.println("信息修改成功！");
                            System.out.println();
                            return;
                        }
                        case 6 ->{
                            System.out.println("请输入职务：");
                            String newPosition = scanner.nextLine();
                            employee.setPosition(newPosition);
                            System.out.println("信息修改成功！");
                            System.out.println();
                            return;
                        }
                        case 7->{
                            System.out.println("请输入工资：");
                            double newSalary = scanner.nextDouble();
                            scanner.nextLine(); // 清除输入缓冲区的换行符
                            employee.setSalary(newSalary);
                            System.out.println("信息修改成功！");
                            System.out.println();
                            return;
                        }

                        case 8->{
                            System.out.println("请输入住址：");
                            String newAddress = scanner.nextLine();
                            employee.setAddress(newAddress);
                            System.out.println("信息修改成功！");
                            System.out.println();
                            return;
                        }

                        case 9->{
                            System.out.println("请输入电话：");
                            String newPhoneNumber = scanner.nextLine();
                            employee.setPhoneNumber(newPhoneNumber);
                            System.out.println("信息修改成功！");
                            System.out.println();
                            return;
                        }
                    }
                }
            }
            System.out.println("未找到该职工");
    }

    // 将员工信息保存到文件
    public void saveToFile (){
            try {
                System.out.println("请输入保存的文件名：");
                String  fileName = scanner.nextLine();
                FileOutputStream fileOutputStream = new FileOutputStream(fileName);
                ObjectOutputStream objectOutputStream = new ObjectOutputStream(fileOutputStream);
                objectOutputStream.writeObject(employees);
                objectOutputStream.close();
                fileOutputStream.close();
                System.out.println("数据保存成功！");
                System.out.println();
            } catch (IOException e) {
                System.out.println("保存失败：" + e.getMessage());
                System.out.println();
            }
        }

    // 从文件读取员工信息
    public void openFromFile (){
            try {
                System.out.println("请输入打开的文件名：");
                String fileName = scanner.nextLine();
                FileInputStream fileInputStream = new FileInputStream(fileName);
                ObjectInputStream objectInputStream = new ObjectInputStream(fileInputStream);
                employees = (List<Employee>) objectInputStream.readObject();
                objectInputStream.close();
                fileInputStream.close();
                System.out.println("文件打开成功！");
                System.out.println();
            } catch (IOException | ClassNotFoundException e) {
                System.out.println("打开失败：" + e.getMessage());
                System.out.println();
            }
        }
}

File: IdeaProjects/Kotlin/Employee information management system/src/EmployeeManagementSystem.kt
import java.io.*
import java.util.*

class EmployeeManagementSystem  // 员工管理系统构造函数
{
    private var employees: MutableList<Employee> = ArrayList() // 员工列表
    var scanner = Scanner(System.`in`)

    // run方法，用于运行员工管理系统
    fun run() {
        // 循环处理用户的操作选择
        var isRunning = true
        while (isRunning) {
            println("请选择操作：")
            println(" 1 ：录入职工信息")
            println(" 2 ：显示所有职工信息")
            println(" 3 ：查询职工信息")
            println(" 4 ：删除职工信息")
            println(" 5 ：修改职工信息")
            println(" 6 ：保存数据到文件")
            println(" 7 ：从文件打开数据")
            println(" 0 ：退出系统")
            println("请输入你的操作：")
            val choice = scanner.nextInt()
            scanner.nextLine() // 清除输入缓冲区的换行符
            when (choice) {
                0 -> isRunning = false
                1 -> addEmployee()
                2 -> displayAllEmployees()
                3 -> searchEmployee()
                4 -> deleteEmployee()
                5 -> updateEmployee()
                6 -> saveToFile()
                7 -> openFromFile()
                else -> println("无效的选择！")
            }
        }
    }

    // 添加员工
    fun addEmployee() {
        println("开始录入职工信息")
        println("请输入职工号：")
        val id = scanner.nextLine()
        for (employee in employees) {
            if (employee.id == id) {
                println("职工号已存在！")
                return
            }
        }
        println("请输入姓名：")
        val name = scanner.nextLine()
        println("请输入性别：")
        val gender = scanner.nextLine()
        println("请输入出生年月：")
        val birthDate = scanner.nextLine()
        println("请输入学历：")
        val education = scanner.nextLine()
        println("请输入职务：")
        val position = scanner.nextLine()
        println("请输入工资：")
        val salary = scanner.nextDouble()
        scanner.nextLine() // 清除输入缓冲区的换行符
        println("请输入住址：")
        val address = scanner.nextLine()
        println("请输入电话：")
        val phoneNumber = scanner.nextLine()
        val employee = Employee(id, name, gender, birthDate, education, position, salary, address, phoneNumber)
        employees.add(employee)
        println("职工信息录入成功！")
        println()
    }

    // 显示所有员工信息
    fun displayAllEmployees() {
        println("所有职工信息如下：")
        for (employee in employees) {
            employee.displayInfo()
            println()
        }
        println()
    }

    // 搜索员工
    fun searchEmployee() {
        println("请输入要查询的职工号：")
        val id = scanner.nextLine()
        for (employee in employees) {
            if (employee.id == id) {
                employee.displayInfo()
                println()
                return
            }
        }
        println("未找到该职工")
        println()
    }

    // 删除员工
    fun deleteEmployee() {
        println("请输入要删除的职工号：")
        val id = scanner.nextLine()
        for (employee in employees) {
            if (employee.id == id) {
                employees.remove(employee)
                println("职工删除成功")
                println()
                return
            }
        }
        println("未找到该职工")
        println()
    }

    // 更新员工信息
    fun updateEmployee() {
        println("请输入要修改的职工号：")
        val id = scanner.nextLine()
        for (employee in employees) {
            if (employee.id == id) {
                println("请选择要修改的信息：")
                println("1.职工号")
                println("2.姓名")
                println("3.性别")
                println("4.出生年月")
                println("5.学历")
                println("6.职务")
                println("7.工资")
                println("8，住址")
                println("9.电话")
                val choice = scanner.nextInt()
                scanner.nextLine() // 清除输入缓冲区的换行符
                when (choice) {
                    1 -> {
                        println("请输入职工号：")
                        val newId = scanner.nextLine()
                        employee.id = newId
                        println("信息修改成功！")
                        println()
                        return
                    }

                    2 -> {
                        println("请输入姓名：")
                        val newName = scanner.nextLine()
                        employee.name = newName
                        println("信息修改成功！")
                        println()
                        return
                    }

                    3 -> {
                        println("请输入性别：")
                        val newGender = scanner.nextLine()
                        employee.gender = newGender
                        println("信息修改成功！")
                        println()
                        return
                    }

                    4 -> {
                        println("请输入出生年月：")
                        val newBirthDate = scanner.nextLine()
                        employee.birthDate = newBirthDate
                        println("信息修改成功！")
                        println()
                        return
                    }

                    5 -> {
                        println("请输入学历：")
                        val newEducation = scanner.nextLine()
                        employee.education = newEducation
                        println("信息修改成功！")
                        println()
                        return
                    }

                    6 -> {
                        println("请输入职务：")
                        val newPosition = scanner.nextLine()
                        employee.position = newPosition
                        println("信息修改成功！")
                        println()
                        return
                    }

                    7 -> {
                        println("请输入工资：")
                        val newSalary = scanner.nextDouble()
                        scanner.nextLine() // 清除输入缓冲区的换行符
                        employee.salary = newSalary
                        println("信息修改成功！")
                        println()
                        return
                    }

                    8 -> {
                        println("请输入住址：")
                        val newAddress = scanner.nextLine()
                        employee.address = newAddress
                        println("信息修改成功！")
                        println()
                        return
                    }

                    9 -> {
                        println("请输入电话：")
                        val newPhoneNumber = scanner.nextLine()
                        employee.phoneNumber = newPhoneNumber
                        println("信息修改成功！")
                        println()
                        return
                    }
                }
            }
        }
        println("未找到该职工")
    }

    // 将员工信息保存到文件
    fun saveToFile() {
        try {
            println("请输入保存的文件名：")
            val fileName = scanner.nextLine()
            val fileOutputStream = FileOutputStream(fileName)
            val objectOutputStream = ObjectOutputStream(fileOutputStream)
            objectOutputStream.writeObject(employees)
            objectOutputStream.close()
            fileOutputStream.close()
            println("数据保存成功！")
            println()
        } catch (e: IOException) {
            println("保存失败：" + e.message)
            println()
        }
    }

    // 从文件读取员工信息
    fun openFromFile() {
        try {
            println("请输入打开的文件名：")
            val fileName = scanner.nextLine()
            val fileInputStream = FileInputStream(fileName)
            val objectInputStream = ObjectInputStream(fileInputStream)
            employees = objectInputStream.readObject() as MutableList<Employee>
            objectInputStream.close()
            fileInputStream.close()
            println("文件打开成功！")
            println()
        } catch (e: IOException) {
            println("打开失败：" + e.message)
            println()
        } catch (e: ClassNotFoundException) {
            println("打开失败：" + e.message)
            println()
        }
    }
}